source/fuzz/fuzzer_pass_split_blocks.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <utility>
+#include <vector>
+
+#include "source/fuzz/fuzzer_pass_split_blocks.h"
+#include "source/fuzz/transformation_split_block.h"
+
+namespace spvtools {
+namespace fuzz {
+
+using opt::IRContext;
+
+FuzzerPassSplitBlocks::FuzzerPassSplitBlocks(
+    opt::IRContext* ir_context, FactManager* fact_manager,
+    FuzzerContext* fuzzer_context,
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, fact_manager, fuzzer_context, transformations) {}
+
+FuzzerPassSplitBlocks::~FuzzerPassSplitBlocks() = default;
+
+void FuzzerPassSplitBlocks::Apply() {
+  // Collect up all the blocks in the module.
+  std::vector<opt::BasicBlock*> available_blocks;
+  for (auto& function : *GetIRContext()->module()) {
+    for (auto& block : function) {
+      available_blocks.push_back(&block);
+    }
+  }
+
+  while (!available_blocks.empty()) {
+    auto block_index = GetFuzzerContext()->GetRandomGenerator()->RandomUint32(
```

Comment:
The intent of the code was to consider the blocks in the module in a random order, randomly deciding for each one whether to split it.  This was achieved by starting with all the block pointers in a vector and on each loop iteration erasing one element of the vector (note that as I had no used a range with erase, it was just one element getting erased).

The reason for doing it this way was to avoid introducing bias by visiting the blocks in the order they appear in the module.  This is important when considering a set of possible transformations when applying one transformation might disable another transformation.

However, having thought about this more I am confident that splitting one block can never make it illegal to split another block, so it's fine to visit the blocks in order and decide randomly for each one whether to split it.

I've changed the code to do that, which makes it simpler.